src/components/ProductCard/ProductCard.jsx

```diff
@@ -1,33 +1,61 @@
 import React from "react";
 import "./ProductCard.css";
+import {useCart} from "../../context/Cart-Context"
+import axios from "axios";
+import { useAuth } from "../../context/Auth-Context";
+import { Link, useNavigate } from "react-router-dom";
+
+
+function ProductCard({ item }) {
+
+  const {itemsInCart,setItemsInCart} = useCart();
+  const {state:{encodedToken}} = useAuth();
```

Comment:
if possible you can destructure `encodedToken` in the context itself and pass in value.

Reply:
in Context its destructured